### ISSUE TITLE
Segmentation fault with default options

### ISSUE THREAD

**Issue by araruna** [glmark2/glmark2#168]
I installed glmark2 in my Manjaro system using the supplied version from AUR, which is labeled as `2021.02-2`.

When I ran the test by typing just `glmark2`, it ran for a while and then closed abruptly. The following is the console output:
```text
=======================================================
    glmark2 2021.02
=======================================================
    OpenGL Information
    GL_VENDOR:     NVIDIA Corporation
    GL_RENDERER:   NVIDIA GeForce GTX 960/PCIe/SSE2
    GL_VERSION:    4.6.0 NVIDIA 470.86
=======================================================
[build] use-vbo=false: FPS: 8638 FrameTime: 0.116 ms
[build] use-vbo=true: FPS: 11722 FrameTime: 0.085 ms
[texture] texture-filter=nearest: FPS: 10380 FrameTime: 0.096 ms
[texture] texture-filter=linear: FPS: 10395 FrameTime: 0.096 ms
[texture] texture-filter=mipmap: FPS: 10462 FrameTime: 0.096 ms
[shading] shading=gouraud: FPS: 10236 FrameTime: 0.098 ms
[shading] shading=blinn-phong-inf: FPS: 10279 FrameTime: 0.097 ms
[shading] shading=phong: FPS: 10034 FrameTime: 0.100 ms
[shading] shading=cel: FPS: 10227 FrameTime: 0.098 ms
[bump] bump-render=high-poly: FPS: 7792 FrameTime: 0.128 ms
[bump] bump-render=normals: FPS: 11627 FrameTime: 0.086 ms
Error: Failed to add fragment shader from file None:
Error:   0(15) : error C0000: syntax error, unexpected identifier, expecting "::" at token "highp"
Error: 0(36) : warning C1503: undefined variable "TextureCoord"
Error: 0(37) : warning C1503: undefined variable "TextureCoord"
Error: 0(38) : warning C1503: undefined variable "TextureCoord"
Error: 
[bump] bump-render=height: Set up failed
[effect2d] kernel=0,1,0;1,-4,1;0,1,0;: FPS: 8544 FrameTime: 0.117 ms
[effect2d] kernel=1,1,1,1,1;1,1,1,1,1;1,1,1,1,1;: FPS: 5432 FrameTime: 0.184 ms
[pulsar] light=false:quads=5:texture=false: FPS: 10355 FrameTime: 0.097 ms
[desktop] blur-radius=5:effect=blur:passes=1:separable=true:windows=4: FPS: 3761 FrameTime: 0.266 ms
[desktop] effect=shadow:windows=4: FPS: 6598 FrameTime: 0.152 ms
[buffer] columns=200:interleave=false:update-dispersion=0.9:update-fraction=0.5:update-method=map: FPS: 1710 FrameTime: 0.585 ms
[buffer] columns=200:interleave=false:update-dispersion=0.9:update-fraction=0.5:update-method=subdata: FPS: 1934 FrameTime: 0.517 ms
[buffer] columns=200:interleave=true:update-dispersion=0.9:update-fraction=0.5:update-method=map: FPS: 2003 FrameTime: 0.499 ms
[ideas] speed=duration: FPS: 8855 FrameTime: 0.113 ms
[jellyfish] <default>: FPS: 7167 FrameTime: 0.140 ms
Error: Failed to add fragment shader from file None:
Error:   0(26) : warning C7022: unrecognized profile specifier "highp"
Error: 0(26) : error C0502: syntax error at token "highp"
Error: 
Error: Failed to add fragment shader from file None:
Error:   0(75) : warning C7022: unrecognized profile specifier "highp"
Error: 0(75) : error C0502: syntax error at token "highp"
Error: 
[terrain] <default>:/usr/include/c++/11.1.0/bits/stl_vector.h:1045: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = int; _Alloc = std::allocator<int>; std::vector<_Tp, _Alloc>::reference = int&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__n < this->size()' failed.
Aborted (core dumped)
```

I've uploaded the core file to be [accessible via this link](https://mega.nz/file/DhpwTZ4S#-5UKZ8gt5hGauIcOZqET0eKgTJwwdAgUvRAz5_WGRPQ).

**Comment by araruna** [comment]
I ran the tool a second time passing `-d` and the attached file contains the output. The new core file [is in this link](https://mega.nz/file/O4xwnCAL#AB-qCeiroPnCk-ljnTVRapcm0CmP-gpec6HOtODxfYQ).

[glmark-debug.txt](https://github.com/glmark2/glmark2/files/7720506/glmark-debug.txt)

**Comment by afrantzis** [comment]
Hi! This seems to be a duplicate of #153, which has been fixed in master and in the fresh 2021.12 release. Closing, but feel free to reopen if you still have issues with the updates.
